turbopack/crates/turbopack-core/src/module_graph/mod.rs

```diff
@@ -272,39 +302,55 @@ impl SingleModuleGraph {
             ExpandAndVisit,
         }
 
-        let mut stack: Vec<(ReverseTopologicalPass, Option<NodeIndex>, NodeIndex)> =
-            vec![(ReverseTopologicalPass::ExpandAndVisit, None, entry_node)];
+        #[allow(clippy::type_complexity)] // This is a temporary internal structure
+        let mut stack: Vec<(
+            ReverseTopologicalPass,
+            Option<(NodeIndex, EdgeIndex)>,
+            NodeIndex,
+        )> = vec![(ReverseTopologicalPass::ExpandAndVisit, None, entry_node)];
         let mut expanded = HashSet::new();
         while let Some((pass, parent, current)) = stack.pop() {
             match pass {
                 ReverseTopologicalPass::Visit => {
                     visit_postorder(
-                        (
-                            parent.map(|parent| graph.node_weight(parent).unwrap()),
-                            graph.node_weight(current).unwrap(),
-                        ),
+                        parent.map(|parent| {
+                            (
+                                graph.node_weight(parent.0).unwrap(),
+                                graph.edge_weight(parent.1).unwrap(),
+                            )
+                        }),
+                        graph.node_weight(current).unwrap(),
                         state,
                     );
                 }
                 ReverseTopologicalPass::ExpandAndVisit => {
                     let action = visit_preorder(
-                        (
-                            parent.map(|parent| graph.node_weight(parent).unwrap()),
-                            graph.node_weight(current).unwrap(),
-                        ),
+                        parent.map(|parent| {
+                            (
+                                graph.node_weight(parent.0).unwrap(),
+                                graph.edge_weight(parent.1).unwrap(),
+                            )
+                        }),
+                        graph.node_weight(current).unwrap(),
                         state,
                     );
                     stack.push((ReverseTopologicalPass::Visit, parent, current));
                     if expanded.insert(current) && action == GraphTraversalAction::Continue {
-                        stack.extend(
-                            graph
-                                .neighbors(current)
-                                // .collect::<Vec<_>>()
-                                // .rev()
-                                .map(|child| {
-                                    (ReverseTopologicalPass::ExpandAndVisit, Some(current), child)
-                                }),
-                        );
+                        let mut walker = graph.neighbors(current).detach();
+                        let neighbors = {
+                            let mut neighbors = vec![];
```

Comment:
That temp vec looks unnecessary. Could we just directly push to the `stack`, or use `walker.map()` if that exists?